CMakeLists.txt

```diff
@@ -413,23 +418,30 @@ endif(QUDA_MAGMA)
 if(QUDA_ARPACK)
   add_definitions(-DARPACK_LIB)
   find_package(PkgConfig REQUIRED)
-  if(QUDA_QMP OR QUDA_MPI)
+
```

Comment:
Can you explain why the changes to CMake here with respect to `ARPACK`?

Reply:
In the old code, if QMP or MPI was deleted, only `libparpack.a` would be linked. ARPACK needs both `libparpack.a` and `libarpack.a` to work with MPI, so we always link with `libarpack.a`, and additionally link with `libparpack.a` for parallel builds.